### PR TITLE
 Add mypy to backend memory

### PR DESCRIPTION
**Pull request by TimeEngineer**

### CODE REVIEW

**Comment by FirelightFlagboy** [comment]
Note: we have a tracking issue for that https://github.com/Scille/parsec-cloud/issues/3264

File: parsec/backend/memory/sequester.py [hunk 1]
```diff
@@ -28,9 +28,9 @@
 
 
 class MemorySequesterComponent(BaseSequesterComponent):
-    def __init__(self):
-        self._organization_component: "MemoryOrganizationComponent" = None
-        self._vlob_component: "MemoryVlobComponent" = None
+    def __init__(self) -> None:
+        self._organization_component: "Optional[MemoryOrganizationComponent]" = None
+        self._vlob_component: "Optional[MemoryVlobComponent]" = None
```
Comment (FirelightFlagboy) [inline parsec/backend/memory/sequester.py:33]: not required to quote the type since we're including `__future__.annotations`

**Review by FirelightFlagboy — APPROVED** [review]

**Comment by TimeEngineer** [comment]
If the current implementation is correct, nope

File: parsec/backend/memory/factory.py [hunk 2]
```diff
@@ -26,15 +26,17 @@
 
 
 @asynccontextmanager
-async def components_factory(config: BackendConfig, event_bus: EventBus):
+async def components_factory(  # type: ignore[misc]
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply parsec/backend/memory/factory.py:29]: Yes

File: parsec/backend/memory/ping.py [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
+    def __init__(self, send_event: Callable[..., Coroutine[Any, Any, None]]) -> None:
         self._send_event = send_event
 
-    def register_components(self, **other_components):
+    def register_components(self, **other_components: Any) -> None:
         pass
 
-    async def ping(self, organization_id: OrganizationID, author: DeviceID, ping: str) -> None:
+    async def ping(
+        self, organization_id: OrganizationID, author: DeviceID | None, ping: str
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply parsec/backend/memory/ping.py:18]: - Yes
- I saw somewhere that there was an `if author`, maybe for an invited one?

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**